src/qoi/include/grins/hitran.h

```diff
@@ -0,0 +1,154 @@
+//-----------------------------------------------------------------------bl-
+//--------------------------------------------------------------------------
+//
+// GRINS - General Reacting Incompressible Navier-Stokes
+//
+// Copyright (C) 2014-2016 Paul T. Bauman, Roy H. Stogner
+// Copyright (C) 2010-2013 The PECOS Development Team
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the Version 2.1 GNU Lesser General
+// Public License as published by the Free Software Foundation.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc. 51 Franklin Street, Fifth Floor,
+// Boston, MA  02110-1301  USA
+//
+//-----------------------------------------------------------------------el-
+
+#ifndef GRINS_HITRAN_H
+#define GRINS_HITRAN_H
+
+
+// C++
+#include <vector>
+#include <string>
+
+// libMesh
+#include "libmesh/libmesh.h"
+
+namespace GRINS
+{
+
+  //! HITRAN interface object
+  /*!
+    This class provides an interface to utilize data from the HITRAN Spectroscopic Database.
+    Data can be pulled directly from the HITRAN website, or through the HITRAN python API, HAPI.
+
+    Two files must be provided to this class, one with various data members, and one with values
+    of the partition function at various temperatures.
+    
+    The datafile must be formatted as such:\n
+      [linecenter wavenumber]\n
+      [linestrength]\n
+      [gamma_air]\n
+      [gamma_self]\n
+      [lower state energy of transition]\n
+      [temperature coefficient]\n
+      [delta_air]
+
+    The partition sum data must be provided as such:\n
+      comma-separated temperatures\n
+      comma-separated partition function values
+
+    Note the partition function data <b>must</b> explicitly include a value at the given reference temperature,
+    which is currently 296K for HITRAN.
+    
+    The basic units for values are those given by HITRAN, which currently are [cm] and [atm]
+  */
+  class HITRAN
+  {
+  public:
+    //! Constructor
+    /*!
+      @param data_file A file containing values for the require quantities, formatted as described above
+      @param parFunction_file A file containing temperature and partition function values, formatted as described above
+    */
+    HITRAN(const std::string& data_file, const std::string& parFunction_file);
```

Comment:
I've started doing the libMesh spacing for `*` and `&`, e.g. `const std::string &`. Let's do that here too, please.

Also, camel case can die in a fire. Please use all lower case for variable and functionnames, with a `_` separating keywords.